Realm/Realm/Schema/ObjectSchema.cs

```diff
@@ -35,6 +35,16 @@ namespace Realms.Schema
     [DebuggerDisplay("Name = {Name}, Properties = {Count}")]
     public class ObjectSchema : IReadOnlyCollection<Property>
     {
+        // TODO andrea:
+        // 1 - move this to the right place
+        // 2- add documentation
+        public enum ObjectSchemaType : byte
```

Comment:
The name of the enum is a little misleading as this is not the type of the objectschema but rather the type of the object the objectschema represents.

Reply:
Agree. How about one of the following, in order of my personal preference?
1. `RealmObjectType`
1. `BaseObjectType`
1. `ObjectType`

Reply:
Let's go with `ObjectType`.

Reply:
Then how would you like the field called? It's currently called exactly `ObjectType`. We could call it `SchemaType`? Would that be ok?